cmd/clairctl/clair/analyze.go

```diff
@@ -0,0 +1,32 @@
+package clair
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/coreos/clair/api/v1"
+)
+
+//Analyze get Analysis os specified layer
+func Analyze(id string) (v1.LayerEnvelope, error) {
+
+	lURI := fmt.Sprintf("%v/layers/%v?vulnerabilities", uri, id)
+	response, err := http.Get(lURI)
+	if err != nil {
+		return v1.LayerEnvelope{}, fmt.Errorf("analysing layer %v: %v", id, err)
+	}
+
+	defer response.Body.Close()
```

Comment:
Always put the defer immediately after the call that opened the resource (after the error check)

```go
response, err := http.Get(lURI)
if err != nil {
	return v1.LayerEnvelope{}, fmt.Errorf("analysing layer %v: %v", id, err)
} 
defer response.Body.Close()

...
```